mesonbuild/compilers/c.py

```diff
@@ -713,6 +708,9 @@ def has_function(self, funcname, prefix, env, extra_args=None, dependencies=None
         if self.links(templ.format(**fargs), env, extra_args, dependencies):
             return True
 
+        if '#include' in prefix:
+            return False
+
```

Comment:
I am not too sure about this, can we get the other two changes in and look at this separately? I am investigating adding `-fno-builtins` as an alternative solution.